File: src/main/java/upbrella/be/user/dto/response/AllHistoryResponse.java
package upbrella.be.user.dto.response;

import lombok.Builder;
import lombok.Getter;

import java.util.List;

@Getter
@Builder
public class AllHistoryResponse {

    private List<SingleHistoryResponse> histories;

    public static AllHistoryResponse of(List<SingleHistoryResponse> get) {

        return AllHistoryResponse.builder()
                .histories(get)
                .build();
    }
}

File: src/main/kotlin/upbrella/be/user/dto/response/AllHistoryResponse.kt
package upbrella.be.user.dto.response

data class AllHistoryResponse(
    val histories: List<SingleHistoryResponse>
) {
    companion object {
        fun of(get: List<SingleHistoryResponse>): AllHistoryResponse {
            return AllHistoryResponse(get)
        }
    }
}
